### ISSUE TITLE
Ruff rule does not describe multiple rules

### ISSUE THREAD

**Issue by rotu** [astral-sh/ruff#5495]
version = 0.0.272

Flake8 supports providing a partial error code like "E4" to select all error codes starting with "E4" (i.e. E401 and E402)

`ruff . --select E4` runs both E401 and E402.
But `ruff rule E4` describes E401 and does not even mention that "E4" also entails E402.

**Comment by rotu** [comment]
Behavior has changed since #2812 was reported. Not sure if current behavior is intentional.

**Comment by charliermarsh** [comment]
My initial preference would be to error on `ruff rule E4`, rather than show multiple rules, since we don't really have a good format for that. At the very least, it's wrong to show the `E402` docs in that case!

**Comment by rotu** [comment]
@charliermarsh erroring would be good. Even better would be showing the list of rules it expands to (just the ID and short name is plenty useful!)

**Comment by dhruvmanila** [comment]
The behavior got changed in `0.0.247`, earlier we were throwing an error:
```console
❯ pipx run "ruff==0.0.246" rule E4                               
⚠️  ruff is already on your PATH and installed at /Users/dhruv/.local/bin/ruff. Downloading and running anyway.
error: invalid value 'E4' for '<RULE>': unknown rule code

For more information, try '--help'.
```

I think it might be related to the `many-to-one` changes: `git log --grep='^many-to-one' --oneline`

**Comment by charliermarsh** [comment]
@dhruvmanila - Are you taking a look at the lack of an error?

**Comment by dhruvmanila** [comment]
You can look at it if you want, I just thought to perform a quick bisection search to find out when did the behavior changed.

**Comment by charliermarsh** [comment]
I'll take a quick look.
